sonic_data_client/mixed_db_client.go

```diff
@@ -98,42 +127,107 @@ func ParseTarget(target string, paths []*gnmipb.Path) (string, error) {
 	return target, nil
 }
 
-func (c *MixedDbClient) DbSetTable(table string, key string, values map[string]string) error {
+func (c *MixedDbClient) GetTable(table string) (swsscommon.ProducerStateTable) {
 	pt, ok := c.tableMap[table]
 	if !ok {
-		pt = swsscommon.NewProducerStateTable(c.applDB, table)
+		if strings.HasPrefix(table, DASH_TABLE_PREFIX) && c.zmqClient != nil {
+			log.V(2).Infof("Create ZmqProducerStateTable:  %s", table)
+			pt = swsscommon.NewZmqProducerStateTable(c.applDB, table, c.zmqClient)
+		} else {
+			log.V(2).Infof("Create ProducerStateTable:  %s", table)
+			pt = swsscommon.NewProducerStateTable(c.applDB, table)
+		}
+
 		c.tableMap[table] = pt
 	}
+
+	return pt
+}
+
+func CatchException(err *error) {
+    if r := recover(); r != nil {
+        *err = fmt.Errorf("%v", r)
+    }
+}
+
+func ProducerStateTableSetWrapper(pt swsscommon.ProducerStateTable, key string, value swsscommon.FieldValuePairs) (err error) {
+	// convert panic to error
+	defer CatchException(&err)
+	pt.Set(key, value, "SET", "")
+	return
+}
+
+func ProducerStateTableDeleteWrapper(pt swsscommon.ProducerStateTable, key string) (err error) {
+	// convert panic to error
+	defer CatchException(&err)
+	pt.Delete(key, "DEL", "")
+	return
+}
+
+type ActionNeedRetry func() error
+
+func RetryHelper(zmqClient swsscommon.ZmqClient, action ActionNeedRetry) {
+	var retry uint = 0
+	var retry_delay = time.Duration(RETYRY_DELAY_MILLISECOND) * time.Millisecond
```

Comment:
Maybe this value can be configured as well?

Reply:
I think maybe it's not necessary because currently there is no config file for this value and add a new parameter to GNMI service for retry delay seems not necessary.